modules/ethereum/src/lib.rs

```diff
@@ -326,6 +328,25 @@ pub trait PruningStrategy: Default {
 	fn pruning_upper_bound(&mut self, best_number: u64, best_finalized_number: u64) -> u64;
 }
 
+/// ChainTime represents the runtime on-chain time
+pub trait ChainTime: Default {
+	/// Is a header timestamp ahead of the current on-chain time.
+	///
+	/// Check whether 'timestamp' is ahead (i.e greater than) the current on-chain
+	/// time. If so, return 'true', 'false' otherwise.
+	fn header_is_ahead(&self, timestamp: u64) -> bool;
```

Comment:
```suggestion
	fn is_timestamp_ahead(&self, timestamp: u64) -> bool;
```
nit: such name would work a bit better for me, `is_` implies it returns `bool` and it just reads a bit better for me. Since we are passing `timestamp` not `header` I'd also change it too. Not super strong on this though, so just a suggestion.

Reply:
Agreed `is_` is more canonical.